command/job_history_test.go

```diff
@@ -166,3 +166,209 @@ namespace "default" {
 		})
 	}
 }
+
+func TestJobHistoryCommand_Diffs(t *testing.T) {
+	ci.Parallel(t)
+
+	// Start test server
+	srv, _, url := testServer(t, true, nil)
+	defer srv.Shutdown()
+	state := srv.Agent.Server().State()
+
+	// Create a job with multiple versions
+	job := mock.Job()
+
+	job.ID = "test-job-history"
+	job.TaskGroups[0].Count = 1
+	must.NoError(t, state.UpsertJob(structs.MsgTypeTestSetup, 1000, nil, job))
+
+	v2 := job.Copy()
+	v2.TaskGroups[0].Count = 2
+	v2.TaggedVersion = &structs.JobTaggedVersion{
+		Name:        "example-tag",
+		Description: "example-description",
+	}
+	must.NoError(t, state.UpsertJob(structs.MsgTypeTestSetup, 1001, nil, v2))
+
+	v3 := job.Copy()
+	v3.TaskGroups[0].Count = 3
+	must.NoError(t, state.UpsertJob(structs.MsgTypeTestSetup, 1002, nil, v3))
+
+	v4 := job.Copy()
+	v4.TaskGroups[0].Count = 4
+	must.NoError(t, state.UpsertJob(structs.MsgTypeTestSetup, 1003, nil, v4))
+
+	t.Run("Without diffs", func(t *testing.T) {
+		ui := cli.NewMockUi()
+		cmd := &JobHistoryCommand{Meta: Meta{Ui: ui}}
+
+		code := cmd.Run([]string{"-address", url, job.ID})
+		must.Zero(t, code)
+
+		out := ui.OutputWriter.String()
+		// There should be four outputs
+		must.Eq(t, 4, strings.Count(out, "Version"))
+		must.Eq(t, 0, strings.Count(out, "Diff"))
+	})
+	t.Run("With diffs", func(t *testing.T) {
+		ui := cli.NewMockUi()
+		cmd := &JobHistoryCommand{Meta: Meta{Ui: ui}}
+
+		code := cmd.Run([]string{"-p", "-address", url, job.ID})
+		must.Zero(t, code)
+
+		out := ui.OutputWriter.String()
+		rawBlocks := strings.Split(out, "Version")
+		// trim empty blocks from whitespace in the output
+		var blocks []string
+		for _, block := range rawBlocks {
+			trimmed := strings.TrimSpace(block)
+			if trimmed != "" {
+				blocks = append(blocks, trimmed)
+			}
+		}
+
+		// Check that we have 4 versions
+		must.Eq(t, 4, len(blocks))
```

Comment:
```suggestion
		must.Len(t, 4, blocks)
```

Same for the other tests.